promise.go

```diff
@@ -85,3 +85,34 @@ func (p *Promise) Result() *Value {
 	val := &Value{ptr, p.ctx}
 	return val
 }
+
+// Then invokes the given function when the promise has been fulfilled, not rejected.
+// The returned Promise resolves after the given function has finished execution.
+func (p *Promise) Then(cb FunctionCallback) *Promise {
+	p.ctx.register()
+	defer p.ctx.deregister()
+	cbID := p.ctx.iso.registerCallback(cb)
+	ptr := C.PromiseThen(p.ptr, C.int(cbID))
+	return &Promise{&Object{&Value{ptr, p.ctx}}}
+}
+
+// Then2 invokes one of the given functions when the promise is fulfilled or rejected.
+// The returned Promise resolves after the callback has finished execution.
+func (p *Promise) Then2(onFulfilled, onRejected FunctionCallback) *Promise {
```

Comment:
`Then2` is an odd API. Can we use a variadic function and do checks for one or two FunctionCallback passed in? ie:

```suggestion
func (p *Promise) Then(cbs ...FunctionCallback) *Promise {
```